### PR TITLE
Format queries/mutations for comments to work with GraphCache

### PR DESCRIPTION
**Pull request by CharlieMcVicker**
Jira: [Comments only sometimes work](https://dailp.atlassian.net/browse/DP-296?atlOrigin=eyJpIjoiYTllYTJmNGMzYTEzNDM4ZTkwNWJlYWNiN2M5ODNkNDgiLCJwIjoiaiJ9)

Currently, comments are posted successfully, but the UX doesn't update automatically. By changing what we fetch in the queries (there is no change to the typescript here) we will get the caching behavior we want. That is, when you post a comment, the word/paragraph comments query will be updated.

This PR should be a great example of making something work with GraphCache!

### CODE REVIEW

File: website/src/graphql/dailp/queries.graphql [hunk 1]
```diff
@@ -427,48 +427,39 @@ query BookmarkedDocuments {
...
+    id
+    displayName
+  }
+  textContent
+  commentType
+}
+
 query WordComments($wordId: UUID!) {
   wordById(id: $wordId) {
+    id
```
Comment (CharlieMcVicker) [inline website/src/graphql/dailp/queries.graphql:451]: We always need to fetch IDs if we want this to work

File: website/src/graphql/dailp/queries.graphql [hunk 2]
```diff
@@ -427,48 +427,39 @@ query BookmarkedDocuments {
...
+      formattedDate
+    }
+  }
+  postedBy {
+    id
+    displayName
+  }
+  textContent
+  commentType
+}
```
Comment (CharlieMcVicker) [inline website/src/graphql/dailp/queries.graphql:447]: These common fields used by two queries and the post mutation are now in a fragment.

File: website/src/graphql/dailp/queries.graphql [hunk 3]
```diff
@@ -427,48 +427,39 @@ query BookmarkedDocuments {
...
-        }
-      }
-      postedBy {
-        id
-        displayName
-      }
-      textContent
-      commentType
+      ...CommentFields
     }
```
Comment (CharlieMcVicker) [inline website/src/graphql/dailp/queries.graphql:454]: The fragment replaces these fields here

File: website/src/graphql/dailp/queries.graphql [hunk 4]
```diff
@@ -603,10 +594,16 @@ mutation PostComment($input: PostCommentInput!) {
...
+      comments {
+        ...CommentFields
+      }
     }
     ... on DocumentParagraph {
       __typename
       id
+      comments {
+        ...CommentFields
+      }
```
Comment (CharlieMcVicker) [inline website/src/graphql/dailp/queries.graphql:606]: We fetch the comments and the fields we need when we post, so we have an up-to-date version on the frontend as soon as the mutation finishes.

**Review by CharlieMcVicker — COMMENTED** [review]
Tour of changes

**Review by GracefulLemming — APPROVED** [review]

**Pull request merged**